plugins/glustershd/rest.go

```diff
@@ -0,0 +1,167 @@
+package glustershd
+
+import (
+	"net/http"
+
+	"github.com/gluster/glusterd2/glusterd2/gdctx"
+	restutils "github.com/gluster/glusterd2/glusterd2/servers/rest/utils"
+	"github.com/gluster/glusterd2/glusterd2/transaction"
+	"github.com/gluster/glusterd2/glusterd2/volume"
+	"github.com/gluster/glusterd2/pkg/api"
+	"github.com/gluster/glusterd2/pkg/errors"
+
+	"github.com/gorilla/mux"
+	"github.com/pborman/uuid"
+	log "github.com/sirupsen/logrus"
+)
+
+func validateVolType(vType volume.VolType) bool {
+	if vType != volume.Replicate && vType != volume.Disperse && vType != volume.DistReplicate && vType != volume.DistDisperse {
+		return false
+	}
+
+	return true
+}
+
+func glustershEnableHandler(w http.ResponseWriter, r *http.Request) {
+	// Implement the help logic and send response back as below
+	p := mux.Vars(r)
+	volname := p["name"]
+
+	ctx := r.Context()
+	logger := gdctx.GetReqLogger(ctx)
+
+	//validate volume name
+	v, err := volume.GetVolume(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusNotFound, errors.ErrVolNotFound.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	// validate volume type
+	if !validateVolType(v.Type) {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "Volume Type not supported", api.ErrCodeDefault)
+		return
+	}
+
+	// Transaction which starts self heal daemon on all nodes with atleast one brick.
+	txn := transaction.NewTxn(ctx)
+	defer txn.Cleanup()
+
+	//Lock on Volume Name
+	lock, unlock, err := transaction.CreateLockSteps(volname)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusInternalServerError, err.Error(), api.ErrCodeDefault)
+		return
+	}
+
+	v.HealEnabled = true
```

Comment:
@prashanthpai  What I am thinking is, if let's say the transaction  UpdateVolInfo fails then it will not affect the volinfo that we retrieve from etcd server(so it's harmless I guess, since we will be using the volinfo from etcd server when initiating heal). If transaction selfheal-start fails then it should rollback the UpdateVolinfo transaction which again is fine because the volinfo doesn't change. Is there something wrong in this approach ?

Reply:
volgen uses information from volinfo from store. (in storeVolume). I have to see if I can modify volgen to use temp volinfo instead of store volinfo.

For now rollback func can unset this flag.

Reply:
> For now rollback func can unset this flag.

+1 

@vpandey-RH Can you update the PR as suggested by @aravindavk